math/check_perfect_square.cpp

```diff
@@ -0,0 +1,117 @@
+/**
+ * @file
+ * @brief This program aims at checking that the given number is perfect square or not
+ * @author [Aman Deep](https://github.com/AmanDeep-21)
+ */
+
+#include <cassert>          // for assert
+#include <iostream>         // for io operations
+
+/**
+ * @brief Getting the integral square root of the number using binary search
+ * @param num number to be checked
+ * @returns int integral part of the square root of @param 'num'
+ */
+int get_integral_square_root(int num){
```

Comment:
Consider using `uint64_t` or `int64_t` rather than `int` ([reference](https://en.cppreference.com/w/cpp/types/integer)). Check other parts of the code.